src/extras/primitives/registerPrimitive.js

```diff
@@ -75,6 +69,33 @@ module.exports = function registerPrimitive (name, definition) {
         }
       },
 
+      applyDefaultComponents: {
+        value: function () {
+          var self = this;
+          var defaultData = cloneObject(this.defaultAttributes);
+
+          // Apply default components.
+          Object.keys(defaultData).forEach(function applyDefault (componentName) {
+            var componentData = defaultData[componentName];
+
+            // Set component properties individually to not overwrite user-defined components.
+            if (componentData instanceof Object && Object.keys(componentData).length) {
+              Object.keys(componentData).forEach(function setProperty (propName) {
```

Comment:
I would add it to our linter https://github.com/aframevr/aframe/issues/1033

But I don't want that to catch in our test suite.